portfolyo/core/shared/plot.py

```diff
@@ -127,56 +186,61 @@ def plot(self: PfState) -> plt.Figure:
         plt.Figure
             The figure object to which the series was plotted.
         """
-        gridspec = {"width_ratios": [1, 1], "height_ratios": [4, 1]}
-        fig, axes = plt.subplots(
-            2, 2, sharex=True, gridspec_kw=gridspec, figsize=(10, 6)
+        gridspec = {"width_ratios": [1, 1, 1], "height_ratios": [4, 1]}
+        fig, (volumeaxes, priceaxes) = plt.subplots(
+            2, 3, sharex=True, sharey="row", gridspec_kw=gridspec, figsize=(10, 6)
         )
-        axes = axes.flatten()
-        axes[0].sharey(axes[1])
-
-        # If freq is MS or longer: use categorical axes. Plot volumes in MWh.
-        # If freq is D or shorter: use time axes. Plot volumes in MW.
-        is_category = tools.freq.shortest(self.index.freq, "MS") == "MS"
 
-        # Volumes.
-        if is_category:
-            so, ss = -1 * self.offtakevolume.q, self.sourced.q
-            kwargs = defaultkwargs("q", is_category)
-        else:
-            so, ss = -1 * self.offtakevolume.w, self.sourced.w
-            kwargs = defaultkwargs("w", is_category)
-        vis.plot_timeseries(axes[0], so, **kwargs)
-        vis.plot_timeseries(axes[1], ss, **kwargs)
+        so, ss, usv = (
+            -1 * self.offtakevolume,
+            self.sourced,
+            self.unsourced,
+        )
 
+        so.plot_to_ax(volumeaxes[0], children=children, kind=so.kind, labelfmt="")
+        ss.plot_to_ax(volumeaxes[1], children=children, kind=Kind.VOLUME, labelfmt="")
+        # Unsourced volume.
+        usv.plot_to_ax(volumeaxes[2], kind=Kind.VOLUME, labelfmt="")
         # Procurement Price.
-        vis.plot_timeseries(axes[2], self.pnl_cost.p, **defaultkwargs("p", is_category))
-
-        # Sourced fraction.
-        vis.plot_timeseries(
-            axes[3], self.sourcedfraction, **defaultkwargs("f", is_category)
+        self.pnl_cost.plot_to_ax(priceaxes[0], kind=Kind.PRICE, labelfmt="")
+        self.sourced.plot_to_ax(
+            priceaxes[1], children=children, kind=Kind.PRICE, labelfmt=""
         )
-
-        # Empty.
-
+        # Unsourced price
+        self.unsourced.plot_to_ax(priceaxes[2], kind=Kind.PRICE, labelfmt="")
         # Set titles.
-        axes[0].set_title("Offtake volume")
-        axes[1].set_title("Sourced volume")
-        axes[2].set_title("Procurement price")
-        axes[3].set_title("Sourced fraction")
+        volumeaxes[0].set_title("Offtake volume")
+        volumeaxes[1].set_title("Sourced volume")
+        volumeaxes[2].set_title("Unsourced volume")
+        priceaxes[0].set_title("Procurement price")
+        priceaxes[1].set_title("Sourced price")
+        priceaxes[2].set_title("Unsourced price")
+
+        limits_vol = [ax.get_ylim() for ax in volumeaxes]
```

Comment:
This calculation can be done inside the `set_max_min_limits` function. Then you don't need to pass it as an argument.